File: MATH/additionOfTwoMatrixes.js
// This function adds two matrices together. It takes two parameters:
// matrixA and matrixB. Both parameters should be 2D arrays representing
// matrices. The function throws an error if the matrices have different
// dimensions (i.e., they don't have the same number of rows and columns).
// If the matrices are valid, the function returns a new matrix that is the
// sum of matrixA and matrixB.
/**
 * 
 *  @param {Array<Array<number>>} matrixA - The first matrix to be subtracted.
 * @param {Array<Array<number>>} matrixB - The matrix to subtract from matrixA.
 */
const addTwoMatrices = (matrixA, matrixB) => {

   // Check if the matrices have the same dimensions.
   // If they don't, throw an error.
   if (matrixA.length !== matrixB.length || matrixA[0].length !== matrixB[0].length) {
      throw new Error("Matrices must have the same dimensions");
   }

   // Initialize an empty array to store the resulting matrix.
   let result = [];

   // Loop through each row of matrixA.
   for (let i = 0; i < matrixA.length; i++) {

      // Initialize an empty array to store the resulting row.
      result[i] = [];

      // Loop through each element in the row.
      for (let j = 0; j < matrixA[i].length; j++) {

         // Calculate the sum of the corresponding elements in matrixA and matrixB.
         // Store this sum in the resulting matrix.
         result[i][j] = matrixA[i][j] + matrixB[i][j];
      }
   }

   // Return the resulting matrix.
   return result;
}

// Example matrices
let matrixA = [
   [1, 2, 3],
   [4, 5, 6],
   [7, 8, 9]
];

let matrixB = [
   [9, 8, 7],
   [6, 5, 4],
   [3, 2, 1]
];

console.log(addTwoMatrices(matrixA, matrixB));

File: MATH/additionOfTwoMatrixes.py
def add_two_matrices(matrix_A, matrix_B):
    """
    This function adds two matrices together and returns the result.
    It first checks if the matrices are the same size. If not, it raises a ValueError.
    Then it initializes an empty list to store the result.
    It uses two nested loops to iterate over the rows and columns of the matrices.
    For each element, it adds the corresponding elements from matrix_A and matrix_B.
    The result is appended to the result list.
    Finally, it prints the result.
    """
    # Check if the matrices are the same size
    if len(matrix_A) != len(matrix_B) or len(matrix_A[0]) != len(matrix_B[0]):
        raise ValueError("The matrices are not the same size")

    # Initialize an empty list to store the result
    result = []

    # Iterate over the rows of matrix_A
    for i in range(len(matrix_A)):
        # Initialize an empty list to store the current row
        row = []

        # Iterate over the columns of the current row
        for j in range(len(matrix_A[0])):
            # Add the corresponding elements from matrix_A and matrix_B
            sum_element = matrix_A[i][j] + matrix_B[i][j]
            # Append the sum to the current row
            row.append(sum_element)

        # Append the current row to the result list
        result.append(row)

    # return the result
    return result

      
   
matrixA = [
   [1, 2, 3],
   [4, 5, 6],
   [7, 8, 9]
]

matrixB = [
   [9, 8, 7],
   [6, 5, 4],
   [3, 2, 1]
]

print(add_two_matrices(matrixA, matrixB))